Modules/_posixsubprocess.c

```diff
@@ -685,6 +685,12 @@ do_fork_exec(char *const exec_array[],
         assert(preexec_fn == Py_None);
 
         pid = vfork();
+        if (pid == -1) {
+            /* If vfork() fails, fall back to using fork(). When it isn't
+             * allowed from a process (such as pid==1 init), vfork can return
+             * -1 with errno EINVAL. https://bugs.python.org/issue47151. */
```

Comment:
It seems like the comment and Changelog entry is not exact: https://bugs.python.org/issue47151#msg416416

Reply:
thanks, reworded.